src/core/sync/condition.d

```diff
@@ -176,14 +201,27 @@ class Condition
      *  SyncError on error.
      */
     void wait()
+    {
+        wait_();
+    }
+
+    /// ditto
+    void wait() shared
```

Comment:
This overload could simply cast away shared and Call the existing wait method.
There is no need for a template.

Ditto for the other methods

Reply:
I updated to use casts instead of overloads except constructors.